flow/record/packer.py

```diff
@@ -107,6 +111,10 @@ def unpack_obj(self, t, data):
         if subtype == RECORD_PACK_TYPE_RECORD:
             identifier, values = value
             identifier = identifier_to_str(identifier)
+
+            if identifier not in self.descriptors:
+                raise RecordDescriptorNotFound(f"No RecordDescriptor found for: {identifier}")
+
             desc = self.descriptors[identifier]
```

Comment:
```suggestion
            desc = self.descriptors.get(identifier)
            if not desc:
                raise RecordDescriptorNotFound(f"No RecordDescriptor found for: {identifier}")
```
This might be more efficient? As this is a very hot code path.